packages/synapse-react-client/src/components/TableColumnSchemaEditor/ColumnModelFormFields/RestrictedValuesField.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useMemo, useState } from 'react'
+import { ColumnTypeEnum } from '@sage-bionetworks/synapse-types'
+import JSONArrayEditorModal from '../../JSONArrayEditor/JSONArrayEditorModal'
+import { getJsonSchemaItemDefinitionForColumnType } from '../TableColumnSchemaEditorUtils'
+import { TextField, TextFieldProps } from '@mui/material'
+
+export type RestrictedValuesFieldProps = {
+  value?: string[]
+  onChange: (newValue: string[]) => void
+  columnType: ColumnTypeEnum
+  TextFieldProps?: Omit<TextFieldProps, 'value' | 'onChange'>
+}
+
+export default function RestrictedValuesField(
```

Comment:
Strategy for `RestrictedValuesField` is a bit different compared to `DefaultValueField`.

Since we're using RJSF, change the schema definition based on the ColumnType.